### DISCUSSION TITLE
Will you consider supporting third-party API providers, such as OpenRouter?

### DISCUSSION THREAD

**Discussion by liuba1223** [glowingjade/obsidian-smart-composer#173] [General]
Will you consider supporting third-party API providers, such as OpenRouter?

**Comment by kevin-on** [comment]
To set OpenAI-compatible endpoints, select "Custom (OpenAI Compatible)" for either the chat model or the apply model. At present, we do not support custom embedding models or saving multiple custom model options.

**Comment by glowingjade** [comment]
Hi, this feature has been implemented in https://github.com/glowingjade/obsidian-smart-composer/pull/240 and is now available!
For setup instructions, check out our documentation:
https://github.com/glowingjade/obsidian-smart-composer/wiki/1.4-Advanced-Configuration
